### ISSUE TITLE
Using Private Registry without Authentication fails

### ISSUE THREAD

**Issue by meerkampdvv** [rancher/rancher#31600]
**What kind of request is this (question/bug/enhancement/feature request):**

Bug

**Steps to reproduce (least amount of steps as possible):**

- Setup a internal Registry for airgapped deployments without the need to authenticate. 
- Setup rancher with CATTLE_SYSTEM_DEFAULT_REGISTRY Environment Variable to use the internal Registry
- Create a new Cluster

**Result:**

Node Creation fails with : 
Error: Cannot perform an interactive login from a non TTY device
Process exited with status 1

**Other details that may be helpful:**

in the node log i see the following cmd fail :
`docker login --username --password-stdin`
no username is set. 

version 2.5.5 did not show this behavior.

**Current Workaround:**
When we create a new Cluster we enable the same Private Registry with a user and a password.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):  v2.5.6
- Installation option (single install/HA): HA Helm deployment on 3 rke nodes. `systemDefaultRegistry` value is set to internal Registry.

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): VSphere 
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VM 8 Core/16 GiB

gz#15692

**Comment by mitchellmaler** [comment]
We just ran into the same issue using the vSphere Driver after upgrading to Rancher 2.5.6. The only workaround was to disable private registries. 

```
[INFO] [node-controller-rancher-machine] private registry detected, authenticating node_name to ourproxy.com
rancher-7996b6d87b-7sgbb rancher 2021/03/08 21:46:30 [ERROR] [node] enqueing node pool c-p4b99:node-pool
rancher-7996b6d87b-7sgbb rancher 2021/03/08 21:46:30 [ERROR] error syncing 'c-p4b99/m-m2qzr': handler node-controller: Error: Cannot perform an interactive login from a non TTY device
rancher-7996b6d87b-7sgbb rancher Process exited with status 1
rancher-7996b6d87b-7sgbb rancher : exit status 1, requeuing
```

**Comment by panho66** [comment]
I ran into same issue using AWS EC2 Node template Rancher 2.4.14 a fresh installation.

2021/03/10 22:57:52 [ERROR] NodeController c-q7pj8/m-zhgrq [node-controller] failed with : Error: Cannot perform an interactive login from a non TTY device
Process exited with status 1
: exit status 1

**Comment by sowmyav27** [comment]
**On 2.5.7-rc1 validated the following scenarios:**

Clusters came up fine and the System project images pulled were from the right reg

**Cluster level Private Registry**
- Node driver cluster with Priv reg with no Auth
- Node driver cluster with Priv reg with Auth
- Custom cluster with Priv reg with no Auth
- Custom cluster with Priv reg with Auth

**Global Level Private Registry**
- Node driver cluster with Priva reg with no Auth
- Custom cluster with Priva reg with no Auth

**Other scenarios:**
- OnTag automation runs
- Upgrade from 2.5.6 - DO cluster which failed to come up. Upgrade to 2.5.7-rc1, verified that the node got deleted and recreated.

**Comment by panho66** [comment]
@sowmyav27  Is there a patch for 2.4.14 same issue ?

**Comment by sowmyav27** [comment]
@panho66 Yes. this is the related issue - https://github.com/rancher/rancher/issues/31682. It has been fixed in 2.4.15
